enhancements/opev-0014.md

```diff
@@ -0,0 +1,207 @@
+the epxanded struct lets us add more data that the community has deemed important
+to have.
+
+### Full Spec and New Fields
+
+When fully expanded, the new product object has three data fields: `subcomponents`,
+`hashes` and `identifiers`. We'll go into each but here is an expanded example:
+
+```json
+{
+  "@context": "https://openvex.dev/ns",
+  "@id": "https://openvex.dev/docs/public/vex-6e8cc8c4d0c38e9e9a60949e5dc0279684d8ad5f6711d9a12bb52247b6cc7271",
+  "author": "",
+  "role": "",
+  "timestamp": "2023-01-16T20:41:55.708329108-06:00",
+  "statements": [
+    {
+      "@id": "https://openvex.dev/docs/public/vex-6e8cc8c4d0c38e9e9a60949e5dc0279684d8ad5f6711d9a12bb52247b6cc7271#statement1",
+      "vulnerability": "CVE-2022-39046",
+      "timestamp": "2022-10-11T20:29:19Z",
+      "status": "fixed",
+      "products": [
+        {
+          "@id:": "pkg:oci/miniprow@sha256:74634d9736a45ca9f6e1187e783492199e020f4a5c19d0b1abc2b604f894ac99?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux",
+          "subcomponents": [
+            { "@id": "pkg:apk/wolfi/glibc@2.36-r4" }, 
+            { "@id": "https://spdx.org/spdxdocs/apko/4fa164f9-90ca-42a8-bafe-5ab9d3cc4af9#packageref" },
+            { "@id": "urn:cdx:f08a6ccd-4dce-4759-bd84-c626675d60a7/1#componentA" }
+            {
+              "identifiers": {
+                "cpe23": "cpe:2.3:a:gnu:glibc:2.36.2:*:*:*:*:*:*:*"
+              }
+            },
+            {
+              "hashes": [ "sha-256": "329c82bb9157b5fa51d67dddd91df0c6881f76dcad7cbe19e683800d371bb142" ]
+            },
+            {
+              "identifiers": {
+                "containerFile": "/path/to/file.txt"
+              }
+            },
+          ],
+          "hashes": [
+            {
+              "algorithm": "sha-256",
+              "value": "329c82bb9157b5fa51d67dddd91df0c6881f76dcad7cbe19e683800d371bb142"
+            }
+          ],
+          "identifiers": [
+            {
+              "type": "purl",
+              "locator": "pkg:oci/miniprow@sha256:74634d9736a45ca9f6e1187e783492199e020f4a5c19d0b1abc2b604f894ac99?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
+            },
+            {
+              "type": "cpe23",
+              "locator": "cpe:2.3:a:*:*:*:*:*:*:*...."
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+With the new product object, this proposal adds the following fields to the OpenVEX
+context. Note that all of these would be optional. 
+
+### subcomponents
+
+The subcomponents field is a list of the subcomponents. These have been moved 
+from a string list at the statement level to be a list of `Component`s. In its
+minimal form, the component is still compact enough to just use the identifiers:
+
+```json
+"subcomponents": [
+  { "@id": "pkg:apk/wolfi/glibc@2.36-r4" }
+]
+```
+
+Expanded, it can specify the full elements of the Component context.
```

Comment:
Just for my knowledge: How would we use this new structure to set the subcomponent to a package/component defined with an ID inside an SPDX or CycloneDX SBOM? (to which the vulnerability match may have been associated by the vulnerability scanner, if it was scanning an SBOM)